pkg/process/config/config_test.go

```diff
@@ -125,9 +127,121 @@ func TestOnlyEnvConfig(t *testing.T) {
 	agentConfig, _ = NewAgentConfig("test", "", "")
 	assert.Equal(t, 99, agentConfig.MaxPerMessage)
 
-	os.Setenv("DD_PROCESS_AGENT_MAX_CTR_PROCS_PER_MESSAGE", "1234")
+	_ = os.Setenv("DD_PROCESS_AGENT_MAX_CTR_PROCS_PER_MESSAGE", "1234")
```

Comment:
If we are moving these config settings to `pkg/config`, any reason why would we test them in `pkg/process` and need to create `NewAgentConfig`, can these be tested in `pkg/config` directly?

```
_ = os.Setenv("DD_LOG_LEVEL", "warn")
	_, _ = NewAgentConfig("test", "", "")
	assert.Equal(t, "warn", config.Datadog.GetString("log_level"))
	_ = os.Unsetenv("DD_LOG_LEVEL")
```

Is there a side-effect you are relying on that requires `NewAgentConfig`?

Reply:
> Is there a side-effect you are relying on that requires NewAgentConfig?
Yep. The process agent relies on the side effects from calling `NewAgentConfig`. This is unfortunately how it has always behaved.

These are regression tests to make sure that the settings migration still works with the codepath the process agent uses to initialize itself still works. That's why we decided to put it in `pkg/process/config` as opposed to `pkg/process`.  There are actually already standalone tests in `pkg/config` that test these setting  behave similar, but are designed to test the settings themselves, instead of the side effects from `NewAgentConfig`.

Reply:
Discussed with @just-chillin looks like we can remove some of the assertions here, everything starting with `DD_LOG_LEVEL` and ending with `DD_PROCESS_AGENT_PROCESS_DISCOVERY_INTERVAL` is already covered in `pkg/config`.